crates/core/src/operations/convert_to_delta.rs

```diff
@@ -328,6 +330,24 @@ impl ConvertToDeltaBuilder {
                 subpath = iter.next();
             }
 
+            let batch_builder = ParquetRecordBatchStreamBuilder::new(ParquetObjectReader::new(
+                object_store.clone(),
+                file.clone(),
+            ))
+            .await?;
+
+            // Fetch the stats
+            let parquet_metadata = batch_builder.metadata();
+            let stats = stats_from_parquet_metadata(
+                &IndexMap::from_iter(partition_values.clone().into_iter()),
+                parquet_metadata.as_ref(),
+                -1,
```

Comment:
This shouldn't be -1. You should use the same function I recently added in the write the get the configuration values: `get_num_idx_cols_and_stats_columns`

Reply:
Good catch, revised now.